src/sentry/api_gateway/api_gateway.py

```diff
@@ -46,9 +54,8 @@ def proxy_request_if_needed(
         org_slug = view_kwargs["organization_slug"]
         return proxy_request(request, org_slug)
 
-    if request.path in REGION_PINNED_URLS:
+    if request.resolver_match and request.resolver_match.url_name in REGION_PINNED_URL_NAMES:
```

Comment:
Using route names allows us to match parameterized URLS (like relay details) in constant time. Using paths would require matching with regexp patterns which would be linear at best.